src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -148,13 +154,17 @@ public static class TagHelperBlockRewriter
                         symbolOffset = 0;
                     }
 
-                    attributeValueStartLocation = symbolStartLocation +
-                                                  span.Start +
-                                                  new SourceLocation(absoluteIndex: 1,
-                                                                     lineIndex: 0,
-                                                                     characterIndex: 1);
+                    attributeValueStartLocation = 
+                        span.Start +
+                        symbolStartLocation +
+                        new SourceLocation(absoluteIndex: 1, lineIndex: 0, characterIndex: 1);
+
                     afterEquals = true;
                 }
+                else if (symbol.Type == HtmlSymbolType.WhiteSpace)
+                {
+                    attributeValueStartLocation = SourceLocation.Advance(attributeValueStartLocation, symbol.Content);
+                }
```

Comment:
well, it'll also handle whitespace between the attribute name and the equals sign.

more generally, what does eating the whitespace prior to the name have to do with empty (quoted in valid cases) attribute values?  (and, why did anything work without this block?)